W-21140013 ip filtering gr (#982)
mulesoft/docs-gateway: pdk/1.7/modules/ROOT/pages/policies-pdk-configure-features-ip-filter.adoc

> = Filter IP addresses

Reviewer: ```suggestion
= Filter IP Addresses
```

Reviewer: It would be better to improve this title to optimize for Agentforce. Something like "Filter Client IPs with Flex Gateway PDK" or whatever.

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> Flex Gateway Policy Development Kit (PDK) provides the IP Filter library functions to filter requests based on IP addresses.

Reviewer: Optimized short description suggestion:
Use the Flex Gateway Policy Development Kit (PDK) IP Filter to control request access by source IP. Create allowlists or blocklists using IPv4/IPv6 addresses and CIDR ranges, and evaluate requests with IpFilter.is_allowed in Rust. Enforce least‑privilege network access for APIs and services.

> To filter IP addresses, you must import the `IpFilter` from the pdk crate. Then, create an IP filter to either allow or block specific IPs:

Reviewer: ```suggestion
Import `IpFilter` from the `pdk` crate, then create an IP filter to either allow or block specific IPs:
```

[source,Rust]
----
use pdk::ip_filter::IpFilter;

// Create an allowlist filter (only specified IPs are permitted)
let ip_filter = IpFilter::allow(&["192.168.1.0/24", "10.0.0.1"])?;

// Create a blocklist filter (specified IPs are denied)
let ip_filter = IpFilter::block(&["192.168.1.0/24", "10.0.0.1"])?;
----

The `IpFilter` accepts an array of strings and supports:

* Both IPv4 and IPv6 addresses.
* Individual IPs, for example: `10.0.0.1`.
* Classless Inter-Domain Routing (CIDR) notation for IP ranges. For example: `192.168.1.0/24`, `10.0.0.0/8`.

After creating the filter, use the `is_allowed` method to check if an IP is permitted:

[source,Rust]
----
// Check if an IP is allowed
if ip_filter.is_allowed("192.168.1.100") {
    // IP is permitted, continue processing
    Flow::Continue(())
} else {
    // IP is blocked, return forbidden response
    Flow::Break(Response::new(403).with_body("Forbidden"))
}
----

== IP Filter Configuration Examples

PDK provides the https://github.com/mulesoft/pdk-custom-policy-examples/tree/{template-policies-url-ver-var}/ip-filter[IP Filter Example policy] to demonstrate how to implement IP-based access control in Rust code.

> Within the example policy, see these code sections for further specific configuration details:

Reviewer: ```suggestion
Within the example policy, see these code sections for additional configuration details:
```
